sdk/scala-sdk/src/main/scala/kalix/scalasdk/action/Action.scala

```diff
@@ -153,14 +157,31 @@ object Action {
     }
   }
 }
+
 abstract class Action {
   @volatile
   private var _actionContext: Option[ActionContext] = None
 
+  /**
+   * An ExecutionContext to use when composing Futures inside Actions.
+   *
+   * Note that this ExecutionContext is only available when handling a message. It will throw an exception if accessed
+   * from constructor.
+   */
+  implicit final def executionContext: ExecutionContext = {
+    actionContext("ExecutionContext is only available when handling a message") match {
```

Comment:
but will this work in for example a `flatMap`? shouldn't it be a `lazy val` instead of `def`? Would be good if it worked for all cases except from the constructor.

Reply:
I think it will break for flatMap with another map inside, because that will execute and access the context after the method has returned and the action context cleared.

Is it really important that the ec is from the actor system, could it be a predefined implicit def returning the global ec instead, and always be available with no caveats?

Reply:
The action context is never cleared. See my comment here: https://github.com/lightbend/kalix-jvm-sdk/pull/921#discussion_r862882928

Reply:
The exception message "ExecutionContext is only available when handling a message" is not completely correct. The only case it will not work is in the constructor.

I would prefer that we eagerly set it together with the first or all times we set actionContext. The reason is that I'm not sure this will work if you for example make a `deferredCall.execute.flatMap(result => ...)`

Reply:
Ah, did not remember the context is never cleared, then I think this will be fine, except for calling from the constructor but is there really a scenario where that makes any sense?

Reply:
ok, ignore my comment about eagerly set. The exception message might also make sense for the end user, they don't think about threads (as I do).